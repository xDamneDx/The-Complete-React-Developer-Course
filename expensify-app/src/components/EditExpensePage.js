import React from 'react';

const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            This is from my edit expense page {props.match.params.id}.
        </div>
    )
};

export default EditExpensePage;
