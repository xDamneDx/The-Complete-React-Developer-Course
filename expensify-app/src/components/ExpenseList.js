import React from 'react';
import { connect } from 'react-redux';

const ExpenseList = (props) => (
    <div>
        <h2>Expense List</h2>
        {props.filters.text}
        {props.expenses.length}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseList);
