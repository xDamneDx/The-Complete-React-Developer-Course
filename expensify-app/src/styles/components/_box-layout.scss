.box-layout {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background: url("/images/bg.jpg");
  background-size: cover;
}

.box-layout__box {
  background: fade-out(white, 0.15);
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}
