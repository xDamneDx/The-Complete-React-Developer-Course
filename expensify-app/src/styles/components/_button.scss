.button {
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  display: inline-block;
  padding: $s-size;
  text-decoration: none;
  color: white;
  border: none;
  background: $blue;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}
