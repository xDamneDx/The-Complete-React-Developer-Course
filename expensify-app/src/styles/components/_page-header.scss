.page-header {
  margin-bottom: $l-size;
  padding: $l-size 0;
  background: $off-white;
}

.page-header__actions {
  margin-top: $m-size;
}

.page-header__title {
  font-weight: 300;
  margin: 0;
  span {
    font-weight: 700;
  }
}
