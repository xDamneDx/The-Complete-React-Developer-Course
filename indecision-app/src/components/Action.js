import React from 'react';

const Action = (props) => (
    <div>
        <button onClick={props.handlePick} disabled={!props.hasOptions}>What shoud I do?</button>
    </div>
);

export default Action;
