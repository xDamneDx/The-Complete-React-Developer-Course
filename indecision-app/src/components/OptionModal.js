import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement(document.getElementById('app'));

const OptionModal = (props) => (
    <Modal
        isOpen={!!props.selectedOption}
        onRequestClose={props.handleClearSelectedOption}
        contentLabel="Selected Option"
    >
        <h3>Selected Option</h3>
        {props.selectedOption && <p>{props.selectedOption}</p>}
        <button onClick={props.handleClearSelectedOption}>oK</button>
    </Modal>
);

export default OptionModal;
