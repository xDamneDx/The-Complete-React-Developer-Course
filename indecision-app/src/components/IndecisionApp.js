import React from 'react';
import Header from './Header';
import Action from './Action';
import Options from './Options';
import OptionModal from './OptionModal';
import AddOption from './AddOption';

export default class IndecisionApp extends React.Component {
    state = {
        options: [],
        selectedOption: undefined
    };
    handeleDeleteOptions = () => {
        this.setState(() => ({ options: [] }));
    };
    handleDeleteOption = (optionToRemove) => {
        this.setState((prevState) => ({
            options: prevState.options.filter((option) => optionToRemove !== option)
        }));
    };
    handlePick = () => {
        const randomNum = Math.floor(Math.random() * this.state.options.length);
        const selectedOption = this.state.options[randomNum];
        this.setState(() => ({
            selectedOption
        }))
    };
    handleAddOption = (option) => {
        if (!option) {
            return 'Enter valid value to add option';
        } else if (this.state.options.indexOf(option) > -1) {
            return 'The option already exists'
        };
        this.setState((prevState) => ({ options: prevState.options.concat(option) }));
    };
    handleClearSelectedOption = () => {
        this.setState(() => ({
            selectedOption: undefined
        }))
    }
    componentDidMount() {
        try {
            const json = localStorage.getItem('options');
            const options = JSON.parse(json);
            if (options) {
                this.setState(() => ({ options }));
            };
        } catch (e) {
            // Do nothing at all
        };
    };
    componentDidUpdate(prevProps, prevState) {
        if (prevState.options.length !== this.state.options.length) {
            const json = JSON.stringify(this.state.options);
            localStorage.setItem('options', json);
        };
    };
    render() {
        const title = 'Indecision';
        const subtitle = 'Put your life in the hands of computer';
        return (
            <div>
                <Header 
                    title={title} 
                    subtitle={subtitle} 
                />
                <Action 
                    hasOptions={this.state.options.length > 0} 
                    handlePick={this.handlePick} 
                />
                <Options 
                    options={this.state.options} 
                    handeleDeleteOptions={this.handeleDeleteOptions} 
                    handleDeleteOption={this.handleDeleteOption} 
                />
                <AddOption 
                    handleAddOption={this.handleAddOption} 
                />
                <OptionModal 
                    selectedOption={this.state.selectedOption}
                    handleClearSelectedOption={this.handleClearSelectedOption}
                />
            </div>
        );
    };
};
