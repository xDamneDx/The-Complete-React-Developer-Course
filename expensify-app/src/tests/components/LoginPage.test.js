import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Should render LoginPage component', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button clicked', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});
