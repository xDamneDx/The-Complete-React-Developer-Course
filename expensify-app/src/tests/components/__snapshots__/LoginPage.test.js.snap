// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage component 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
